zebra/label_manager.c

```diff
@@ -138,7 +136,11 @@ assign_label_chunk (label_owner_t owner)
 		else
 				lmc->start = ((struct label_manager_chunk *)
 							  listgetdata(listtail(lbl_mgr.lc_list)))->end + 1;
-		lmc->end = lmc->start + lbl_mgr.chunk_size - 1;
+		if (lmc->start > MPLS_MAX_UNRESERVED_LABEL - size + 1) {
```

Comment:
Ya, I agree (actually, I wrote it this way initially) but I changed it to avoid a (not very likely) integer overflow